packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -26,22 +24,33 @@ import {
   rtlTextContainer,
 } from '../../utils';
 import {
-  ShorthandRenderFunction,
-  ShorthandCollection,
   ComponentEventHandler,
   ObjectShorthandCollection,
   FluentComponentStaticProps,
+  ShorthandRenderFunction,
 } from '../../types';
-import {
-  getAllSelectableChildrenId,
-  isAllGroupChecked,
-  hasSubtree,
-  removeItemAtIndex,
-  getSiblings,
-  TreeContext,
-  TreeRenderContextValue,
-  processItemsForSelection,
-} from './utils';
+import { useSelectableTree } from './hooks/useSelectableTree';
+import { BaseFlatTreeItem } from './hooks/flattenTree';
+
+export interface TreeRenderContextValue {
+  toggleActive: (ids: string[], e: React.SyntheticEvent) => void;
+  toggleSelect: (ids: string[], e: React.SyntheticEvent) => void;
+  focusFirstChild: (itemId: string) => void;
+  focusParent: (itemId: string) => void;
+  siblingsExpand: (e: React.SyntheticEvent, itemProps: TreeItemProps) => void;
+  registerItemRef: (id: string, node: HTMLElement) => void;
+  getItemById: (id: string) => BaseFlatTreeItem & { [key: string]: any };
```

Comment:
```suggestion
  getItemById: (id: string) => BaseFlatTreeItem;
```

I would like to avoid this part if possible

Reply:
So in `useTreeSelectState.tsx`:

```ts 
export interface SelectableFlatTreeItem extends BaseFlatTreeItem {
  selected?: boolean | 'indeterminate';
} 
```

And when I need to use `flatTree[id].selected`, I do `(flatTree[id] as SelectableFlatTreeItem).selected = true`